pkg/metrics/backends/prometheus/prometheus.go

```diff
@@ -93,3 +97,51 @@ func (b *Backend) QueryInstant(ctx context.Context, query string, ts time.Time)
 		return nil, errors.Errorf("unknown supported type: '%q'", v)
 	}
 }
+
+// getAlertmanagers returns Alertmanager clients that is associated with the backend.
+func (b *Backend) getAlertmanagers(ctx context.Context) ([]*amclient.Alertmanager, error) {
+	level.Debug(b.logger).Log("msg", "Get active Alertmanagers")
+	amResult, err := b.prometheus.AlertManagers(ctx)
+	if err != nil {
+		return nil, err
+	}
+	var ams []*amclient.Alertmanager
+	// Get only active Alertmanager instances.
+	for _, a := range amResult.Active {
+		u, _ := url.Parse(a.URL)
```

Comment:
Should handle errors here 'cuz Alertmanager Url can be misconfigured in Prometheus?

Reply:
Ok, actually if there is an invalid URL, a 503 status code error will be returned when we use the client.